test/test.js

```diff
@@ -0,0 +1,86 @@
+process.env.NODE_ENV = 'test';
+
+let mongoose = require("mongoose");
+let Article = require('../models/article');
+
+//Require the dev-dependencies
+let chai = require('chai');
+let chaiHttp = require('chai-http');
+let server = require('../server');
+let should = chai.should();
+
+
+chai.use(chaiHttp);
+//Our parent block
+describe('articles', () => {
+    beforeEach((done) => { //Before each test we empty the database
+        Article.remove({}, (err) => { 
+           done();           
+        });        
+    });
+/*
+  * Test the /GET route
+  */
+  describe('/GET articles', () => {
+      it('it should GET all the articles', (done) => {
+        chai.request(server)
+            .get('/articles')
+            .end((err, res) => {
+                  res.should.have.status(201);
+                  res.body.should.be.a('array');
+                  res.body.length.should.be.eql(0);
+              done();
+            });
+      });
+  });
+
+  /*
+  * Test the /POST route
+  */
+  describe('/POST ', () => {
+      it('it should POST a article ', (done) => {
+          let article = {
+            title : 'learning test',
+            description :  'writing test is always good',
+            author : 'justin',
+            tags :  'hello'
+          }
+        chai.request(server)
+            .post('/articles')
+            .send(article)
+            .end((err, res) => {
+                  res.should.have.status(201);
+                  res.body.should.be.a('object');
+                  res.body.should.have.property('author').eql('justin');
+                  res.body.should.have.property('_id');
+                  res.body.should.have.property('time');
+
+              done();
+            });
+      });
+
+      it('it should fail to POST a article without title', (done) => {
+        let article = {
+          description :  'writing test is always good',
+          author : 'justin',
+          tags :  'hello'
+        }
```

Comment:
@sastaachar , I am not sure how we are taking the request, object 

Is it a object or
```
[title]=
[author]=
```

I guess
this format is good
```
{
author:
title:
}
```

Also take a look why this test is failing , 
it should throw an error

Reply:
I will take a look, if you didn't beat me to it :)

Reply:
test fails at 1) it should fail to POST a article without title

Reply:
title is missing there so an internal error will thrown i.e 400 ,
but there was a typo so it should work now